creator-node/src/routes/nodeSync.js

```diff
@@ -253,146 +275,118 @@ async function _nodesync (req, walletPublicKeys, creatorNodeEndpoint, dbOnlySync
-          const myCnodeEndpoint = await middlewares.getOwnEndpoint(req)
-          userReplicaSet = await middlewares.getCreatorNodeEndpoints(req, fetchedWalletPublicKey)
+      // Build user replica set array for use in file fetch + save
+      // TODO move this logic to right before file save ops...
+      let userReplicaSet = []
+      try {
+        const myCnodeEndpoint = await middlewares.getOwnEndpoint(req)
+        userReplicaSet = await middlewares.getCreatorNodeEndpoints(req, fetchedWalletPublicKey)
 
-          // push user metadata node to user's replica set if defined
-          if (config.get('userMetadataNodeUrl')) userReplicaSet.push(config.get('userMetadataNodeUrl'))
+        // push user metadata node to user's replica set if defined
+        if (config.get('userMetadataNodeUrl')) userReplicaSet.push(config.get('userMetadataNodeUrl'))
 
-          // filter out current node from user's replica set
-          userReplicaSet = userReplicaSet.filter(url => url !== myCnodeEndpoint)
+        // filter out current node from user's replica set
+        userReplicaSet = userReplicaSet.filter(url => url !== myCnodeEndpoint)
 
-          // Spread + set uniq's the array
-          userReplicaSet = [...new Set(userReplicaSet)]
-        } catch (e) {
-          req.logger.error(redisKey, `Couldn't get user's replica sets, can't use cnode gateways in saveFileForMultihash - ${e.message}`)
-        }
+        // Spread + set uniq's the array
+        userReplicaSet = [...new Set(userReplicaSet)]
+      } catch (e) {
+        req.logger.error(redisKey, `Couldn't get user's replica sets, can't use cnode gateways in saveFileForMultihashToFS - ${e.message}`)
       }
 
       if (!walletPublicKeys.includes(fetchedWalletPublicKey)) {
         throw new Error(`Malformed response from ${creatorNodeEndpoint}. Returned data for walletPublicKey that was not requested.`)
       }
-      const fetchedCnodeUserUUID = fetchedCNodeUser.cnodeUserUUID
 
+      const {
+        latestBlockNumber: fetchedLatestBlockNumber,
+        clock: fetchedLatestClockVal,
+        clockRecords: fetchedClockRecords
+      } = fetchedCNodeUser
+
+      // Error if returned data is not within requested range
+      if (fetchedLatestClockVal < localMaxClockVal) {
+        throw new Error(`Cannot sync for localMaxClockVal ${localMaxClockVal} - imported data has max clock val ${fetchedLatestClockVal}`)
+      } else if (fetchedLatestClockVal === localMaxClockVal) {
+        // Already up to date, no sync necessary
+        req.logger.info(redisKey, `User ${fetchedWalletPublicKey} already up to date! Both nodes have latest clock value ${localMaxClockVal}`)
+        continue
+      } else if (localMaxClockVal !== -1 && fetchedClockRecords[0] && fetchedClockRecords[0].clock !== localMaxClockVal + 1) {
+        throw new Error(`Cannot sync - imported data is not contiguous. Local max clock val = ${localMaxClockVal} and imported min clock val ${fetchedClockRecords[0].clock}`)
```

Comment:
mentioned earlier, but how do we reconcile this? whether in this PR or later - seems like this scenario always has some minor chance of occurring. let's discuss offline

Reply:
yep, made a note to chat about this

Reply:
This might be a blocker no? Seems like any user who hits this will break permanently